frontend/src/modules/scaffold/views/scaffoldAssetItemView.js

```diff
@@ -0,0 +1,225 @@
+          break;
+      }
+
+      return baseEditor.nestedForm.fields[this.items].editor;
+    },
+
+    removeDragItems: function() {
+      this.dragItems.forEach(this.removeDragItem, this);
+      this.dragItems = [];
+    },
+
+    addDragItem: function(itemView) {
+      var value = itemView.value;
+      var index = itemView.$el.closest('.list-item').index();
+      var elm = $('<div></div>', {
+        css: {
+          top: parseInt(value[this.topAttr]).toFixed(this.precision)+'%',
+          left: parseInt(value[this.leftAttr]).toFixed(this.precision)+'%',
+          position: 'absolute'
+        },
+        text: index + 1
+      });
+
+      this.dragItems.push({
+        $el: elm,
+        view: itemView
+      });
+
+      elm.appendTo('.scaffold-asset-item-img-holder').draggable({
+        containment: 'parent',
+        stop: this.onDragStop.bind(this)
+      });
+    },
+
+    updateDragItem: function(item) {
+      var index = item.view.$el.closest('.list-item').index();
+      item.$el.css({
+        top: parseInt(item.view.value[this.topAttr]).toFixed(this.precision)+'%',
+        left: parseInt(item.view.value[this.leftAttr]).toFixed(this.precision)+'%',
+      }).text(index + 1);
+    },
+
+    removeDragItem: function(item) {
+      if (!item.$el.hasClass('ui-draggable')) return;
+      item.$el.draggable('destroy').remove();
+    },
+
+    onItemListRemove: function(listView, listItemView) {
+      var index = -1;
+      for (var i = 0; i < this.dragItems.length; i++) {
+        if (this.dragItems[i].view === listItemView) {
+          index = i;
+          break;
+        }
+      }
+      this.removeDragItem(this.dragItems[index]);
+      this.dragItems.splice(index, 1);
+    },
+
+    onItemListAdd: function(listView, listItemView) {
+      this.addDragItem(listItemView);
+    },
+
+    onItemListChange: function(listView) {
+      this.dragItems.forEach(this.updateDragItem, this);
+    },
+
+    onDragStop: function(event, ui) {
+      var $dragHandle = $(event.target);
+      var $parent = this.$('.scaffold-asset-item-img-holder');
+      var left = (parseInt($dragHandle.css("left")) / ($parent.width() / 100)).toFixed(this.precision);
+      var top = (parseInt($dragHandle.css("top")) / ($parent.height() / 100)).toFixed(this.precision);
+      var index = $dragHandle.css({left: left + '%', top: top+'%'}).data('index');
+
+      var child = _.find(this.dragItems, function(i) {
+        return i.$el[0] === event.target;
+      });
+
+      if (!child) return;
+
+      var value = {};
+      value[this.leftAttr] = left;
+      value[this.topAttr] = top;
+      child.view.setValue(_.extend({}, child.view.value, value));
+      child.view.renderSummary();
+    },
+
+    onAssetButtonClicked: function(event) {
+      event.preventDefault();
+
+      Origin.trigger('modal:open', AssetManagementModalView, {
+        collection: new AssetCollection,
+        assetType: 'Asset:image',
```

Comment:
This limits it to image type assets. It would be good to be able to specify in the schema which type of asset can be selected. Alternatively the naming should imply this is image only.

Reply:
On a related note, as part of the Scaffold refactor, I did lay the groundwork to avoid these colon-separated types:
#2156 
#### Old
```json
"inputType": "Asset:image"
```
#### New
```json
"inputType": {
  "type": "Asset",
  "media": "image"
}
```
Asset management does still need tweaking to work with it though, e.g.
https://github.com/adaptlearning/adapt_authoring/blob/32be3c1bb96d7ecef7d4bf46e1bf6b18f36226ef/frontend/src/modules/assetManagement/views/assetManagementModalView.js#L28

Reply:
Yes it is currently only limited to images.
For what other media types would you like to have x/y coordinates @dancgray ?

Reply:
Video would be a possibility @lc-thomasberger

Reply:
Then you would have to handle time. So it would be better to have a new scaffold editor for this.

Reply:
Only if you wanted the items to record a time position as well as x/y. But happy to leave this until we have a more fleshed out requirement for other asset types.

Reply:
In my opinion it does only make sense to have the position dependent on time for videos.